engine/src/org/pentaho/di/kitchen/Kitchen.java

```diff
@@ -77,16 +78,16 @@ public static void main( String[] a ) throws Exception {
     final ExecutorService executor = ExecutorUtil.getExecutor();
     final RepositoryPluginType repositoryPluginType = RepositoryPluginType.getInstance();
 
-    final Future<Pair<KettlePluginException, Future<KettleException>>> repositoryRegisterFuture =
-      executor.submit( new Callable<Pair<KettlePluginException, Future<KettleException>>>() {
+    final Future<Map<KettlePluginException, Future<KettleException>>> repositoryRegisterFuture =
```

Comment:
Pair is a simple struct - Map<KettlePluginException, Future<KettleException>> seems like the wrong thing to use. I suggest:

Map.Entry<KettlePluginException, Future<KettleException>> - - - Map.Entry is a very similar structure to Pair.